h2/src/main/org/h2/command/dml/Select.java

```diff
@@ -847,9 +854,11 @@ public void prepare() {
             sort = prepareOrder(orderList, expressions.size());
             orderList = null;
         }
-        for (int i = 0; i < expressions.size(); i++) {
-            Expression e = expressions.get(i);
-            expressions.set(i, e.optimize(session));
+        if (!isExistsSubquery) {
+            for (int i = 0; i < expressions.size(); i++) {
+                Expression e = expressions.get(i);
+                expressions.set(i, e.optimize(session));
+            }
```

Comment:
I would rather things like "1/0" would fail at prepare time - it seems to me to be hiding potential errors. I can't see any good coming of glossing over stuff like that.

And yeah, I am nervous about this codebase - it's highly optimised and there are hidden intricacies I have not fully mastered